src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt

```diff
@@ -101,36 +102,47 @@ class RuleHeaderPanel : JBPanel<RuleHeaderPanel>(FlowLayout(FlowLayout.LEFT)) {
         updateRuleSeverity(severity)
     }
 
-    fun update(ruleKey: String, type: RuleType, severity: IssueSeverity, finding: Finding) {
+    fun update(project: Project, ruleKey: String, type: RuleType, severity: IssueSeverity, finding: Finding) {
         clear()
         updateCommonFields(type, ruleKey)
         updateRuleSeverity(severity)
         val serverFindingKey: String
         val isIssueValid: Boolean
+        var taintVulnerability: LocalTaintVulnerability
+        var liveIssue: LiveIssue
 
         when (finding) {
             is LiveIssue -> {
                 serverFindingKey = finding.serverFindingKey ?: return
                 isIssueValid = finding.isValid
+                liveIssue = finding
+                listenIssueButton(isIssueValid, project, liveIssue.file, serverFindingKey, false,
+                    liveIssue, null)
             }
 
             is LocalTaintVulnerability -> {
                 serverFindingKey = finding.key() ?: return
                 isIssueValid = finding.isValid()
+                taintVulnerability = finding
+                finding.file()?.let {
+                    listenIssueButton(isIssueValid, project, it, serverFindingKey, true,
+                        null, taintVulnerability)
+                }
             }
 
             else -> {
                 return
             }
         }
-
-        listenIssueButton(isIssueValid)
     }
 
-    private fun listenIssueButton(isValid: Boolean) {
+    private fun listenIssueButton(
+        isValid: Boolean, project: Project, file: VirtualFile, issueKey: String, isTaintVulnerability: Boolean,
+        liveIssue: LiveIssue?, taintVulnerability: LocalTaintVulnerability?,
+    ) {
         changeStatusButton.action = object : AbstractAction(MARK_AS_RESOLVED) {
             override fun actionPerformed(e: ActionEvent?) {
-                System.out.println("Open Dialog")
+                MarkAsResolvedAction().openMarkAsResolvedDialog(project, file, issueKey, isTaintVulnerability, liveIssue, taintVulnerability)
```

Comment:
I know that we did the same for hotspots, but I find it a bit strange to create the action from here. Creating the action is not useful because we don't invoke it. But that should work. Maybe we could make the method static or part of a service?